pkg/image/image.go

```diff
@@ -192,14 +197,18 @@ func Init(path string, writable bool) (*Image, error) {
 		}
 		fileinfo, err := img.File.Stat()
 		if err != nil {
-			img.File.Close()
+			_ = img.File.Close()
 			return nil, err
 		}
 
 		err = rf.format.initializer(img, fileinfo)
+		if err == ErrArchMismatch {
+			_ = img.File.Close()
+			return nil, ErrArchMismatch
+		}
 		if err != nil {
-			sylog.Debugf("%s format initializer returns: %s", rf.name, err)
-			img.File.Close()
+			sylog.Errorf("%s format initializer returned: %s", rf.name, err)
```

Comment:
It will display annoying false error message for each executed initializer until it found the correct format, should be `Debugf` or `Verbosef`